File: problems/1881_maximum_value_after_insertion/solution.py
class Solution:
    def maxValue(self, n: str, x: int) -> str:
        is_neg = n[0] == '-'
        start = 1 if is_neg else 0
        for i in range(start, len(n)):
            digit = int(n[i])
            if is_neg and digit > x:
                return n[:i] + str(x) + n[i:]
            if not is_neg and digit < x:
                return n[:i] + str(x) + n[i:]
        return n + str(x)

File: problems/1881_maximum_value_after_insertion/Solution.java
class Solution {
    public String maxValue(String n, int x) {
        boolean isNeg = n.charAt(0) == '-';
        int start = isNeg ? 1 : 0;
        for (int i = start; i < n.length(); i++) {
            int digit = n.charAt(i) - '0';
            if (isNeg && digit > x) return n.substring(0, i) + x + n.substring(i);
            if (!isNeg && digit < x) return n.substring(0, i) + x + n.substring(i);
        }
        return n + x;
    }
}
